composite/proof-of-reserves/src/balance.ts

```diff
@@ -1,15 +1,30 @@
 import { Execute, Implementations } from '@chainlink/types'
+import amberdata from '@chainlink/amberdata-adapter'
 import blockchainCom from '@chainlink/blockchain.com-adapter'
+import blockchair from '@chainlink/blockchair-adapter'
 import blockcypher from '@chainlink/blockcypher-adapter'
+import btc_com from '@chainlink/btc.com-adapter'
+import cryptoapis from '@chainlink/cryptoapis-adapter'
+import sochain from '@chainlink/sochain-adapter'
 
 export type BitcoinIndexerOptions = { type?: BitcoinIndexer }
 export enum BitcoinIndexer {
+  Amberdata = 'amberdata',
```

Comment:
This mapping between the enum name and the adapter defined `NAME` could pose a problem. I think that enums are too static for our use case here. I think it would be better if we just define `BitcoinIndexer` as string type alias and compute it from the map. Something like this:

```javascript
export type BitcoinIndexerOptions = { type?: BitcoinIndexer }
export type BitcoinIndexer = string

const implLookup: Implementations = {
  [amberdata.NAME]: amberdata,
  [blockchainCom.NAME]: blockchainCom,
  [blockcypher.NAME]: blockcypher,
  [blockchair.NAME]: blockchair,
  [btc_com.NAME]: btc_com,
  [cryptoapis.NAME]: cryptoapis,
  [sochain.NAME]: sochain,
}

const isBitcoinIndexer = (envVar?: string): envVar is BitcoinIndexer =>
  Object.keys(implLookup).includes(envVar as any)

export const getBitcoinIndexer = (): BitcoinIndexer | undefined => {
  const bitcoinIndexer = process.env.BTC_INDEXER_ADAPTER
  return isBitcoinIndexer(bitcoinIndexer) ? (bitcoinIndexer as BitcoinIndexer) : undefined
}

export const getImpl = (options: BitcoinIndexerOptions): Execute => {
  const prefix = options.type?.toUpperCase()
  const impl = options.type && implLookup[options.type]
  if (!impl) throw Error(`Unknown balance adapter type: ${options.type}`)

  return (data) => {
    const config = impl.makeConfig(prefix)
    const execute = impl.makeExecute(config)
    return execute(data)
  }
}
```

Reply:
@HenryNguyen5 would love your input on how to best type this piece of code?